test/functionalTest/cases/3004_v2_reg_create/v2_reg_create_errors.test

```diff
@@ -0,0 +1,1648 @@
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "invalid value of /supportedForwardingMode/",
+    "error": "BadRequest"
+}
+
+
+36. provider/http not a JSON object
+===================================
+HTTP/1.1 400 Bad Request
+Content-Length: 67
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "/http/ must be a JSON object",
+    "error": "BadRequest"
+}
+
+
+37. provider/http/url missing
+=============================
+HTTP/1.1 400 Bad Request
+Content-Length: 72
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "/http/ must contain a field /url/",
+    "error": "BadRequest"
+}
+
+
+38. provider/http/url not a string
+==================================
+HTTP/1.1 400 Bad Request
+Content-Length: 82
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "/url/ field of /http/ must be a JSON string",
+    "error": "BadRequest"
+}
+
+
+39. provider/http/url is the empty string
+=========================================
+HTTP/1.1 400 Bad Request
+Content-Length: 87
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "/url/ field of /http/ cannot be the empty string",
+    "error": "BadRequest"
+}
+
+
+40. provider/http/url invalid
+=============================
+HTTP/1.1 400 Bad Request
+Content-Length: 71
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "/url/ field of /http/ is invalid",
+    "error": "BadRequest"
+}
+
+
+41. status is not a string
+==========================
+HTTP/1.1 400 Bad Request
+Content-Length: 64
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "/status/ must be a string",
+    "error": "BadRequest"
+}
+
+
+42. invalid value for status
+============================
+HTTP/1.1 400 Bad Request
+Content-Length: 65
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "invalid value for /status/",
+    "error": "BadRequest"
+}
+
+
+43. legacyForwarding missing
+============================
+HTTP/1.1 400 Bad Request
+Content-Length: 114
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "the field /legacyForwarding/ must be present as a boolean and set to /true/",
+    "error": "BadRequest"
+}
+
+
+44. legacyForwarding not a boolean
+==================================
+HTTP/1.1 400 Bad Request
+Content-Length: 103
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "description": "the field /legacyForwarding/ must be a boolean and set to /true/",
+    "error": "BadRequest"
+}
+
+
+45. legacyForwarding set to false
```

Comment:
I would suggest a different response:

```
45. legacyForwarding set to false
=================================
HTTP/1.1 501 Not implemented
Content-Length: ...
Content-Type: application/json
Fiware-Correlator: REGEX([0-9a-f\-]{36})
Date: REGEX(.*)

{
    "description": "Only NGSIv1-based forwarding supported at the present moment. Set legacyForwarding to true",
    "error": "NotImplemented"
}
```

Or a similar wording for descriptions.

To implement 501 properly (instead of 400) woud could use the same apporach than for supportedForwardingMode, getting out the logic from parsing layer to the postRegistration() service routine.

Reply:
The same applies to step 43

Reply:
Maybe you could need a way to detect if legacyFowarwind field was provided. In that case, we could use some parallel `xxxxxProvided` boolean. Have a look to SubscrionUpdate.h case, it is plenty of examples.

Reply:
Not sure what you mean. The absence of the legacyForwarding field is detected already, precisely in test case 43.
As an error is returned, no help variable is needed.
Seems I haven't understood what you mean ...
No problem in changing the error texts.

Reply:
So, what you are proposing is that if a user sends a payload including "legacyForwarding=false", the broker should respond with a 501 Not Implemented and at the same time ask the user to use legacyForwarding, but with another value?    Is it implemented or not?

No, this makes no sense.
Much better the way it is right now.

If you really need to put in this temporary 501, it seems to me the only logical solution is to return a 501 if legacyForwarding is present and not give error if it is not present.

Reply:
Postponed for a later PR